index.js

```diff
@@ -1,6 +1,59 @@
-function gampee()
-{
-	throw new Error("Not implemented");
+function gampee(ecData) {
+
+	if (!ecData) {
+		throw new TypeError();
+	}
+
+	if (!Array.isArray(ecData)) {
+		ecData = [ecData];
+	}
+
+	var ecParams = {};
+
+	ecData.forEach(function (dataItem, diIdx) {
+		var dataKeyPrefix = "";
+
+		if (dataItem.type === "impression") {
+			dataKeyPrefix = "il" + diIdx;
+			if (dataItem.list) ecParams[dataKeyPrefix + "nm"] = "" + dataItem.list;
+		}
+
+		if (dataItem.type !== "impression") {
```

Comment:
They will get an error :)

I want to get this first draft out, before I start implementing validations.